### PR TITLE
More weapon funcs

### PR DESCRIPTION
**Pull request by joshlory**
Includes:

* [func_ptr_80170004_w_010](https://decomp.me/scratch/SH0iK)
* [func_ptr_80170008_w_043](https://decomp.me/scratch/LeMEl)
* [EntityWeaponAttack_w_045](https://decomp.me/scratch/4oeHu)

### CODE REVIEW

File: src/weapon/w_010.c [hunk 1]
```diff
@@ -27,7 +27,34 @@ void DebugInputWait(const char* msg) {
...
 
-INCLUDE_ASM("weapon/nonmatchings/w_010", func_ptr_80170004);
+extern AnimationFrame D_4A000_8017ABA0[];
+
+s32 func_ptr_80170004(Entity* self) {
+    if (self->ext.weapon.parent->entityId == 0) {
+        DestroyEntity(self);
+        return;
+    }
+    if (self->ext.weapon.parent->ext.weapon.equipId != 7) {
```
Comment (sozud) [inline src/weapon/w_010.c:37]: ITEM_IRON_SHIELD

File: src/weapon/w_010.c [hunk 2]
```diff
@@ -27,7 +27,34 @@ void DebugInputWait(const char* msg) {
...
+        DestroyEntity(self);
+        return;
+    }
+    FntPrint("charah:%02x\n", self->animSet);
+    FntPrint("oya_ptr:%02x\n", self->ext.weapon.parent);
+    self->posX.i.hi = PLAYER.posX.i.hi;
+    self->posY.i.hi = PLAYER.posY.i.hi;
+    if (self->step == 0) {
+        self->animSet = self->ext.weapon.parent->animSet;
+        self->palette = 0x8194;
```
Comment (sozud) [inline src/weapon/w_010.c:47]: PAL_OVL I think?

**Review by sozud — APPROVED** [review]

File: src/weapon/w_010.c (hunk 2 above)
Comment (joshlory) [reply src/weapon/w_010.c:47]: I don't see it used elsewhere in the weapons folder yet, should I update everywhere?

Comment (sozud) [reply src/weapon/w_010.c:48]: Actually I think PAL_OVL is misnamed. Seems like it's used to specify a type of palette animation?
```
RenderEntities
...
            if (palette & 0x8000) {
                poly->clut = g_ClutIds[palette & 0x7FFF];
            } else {
                poly->clut = g_ClutIds[animFrame[2] + palette];
            }
```

@Xeeynamo what do you think?

Comment (Xeeynamo) [reply src/weapon/w_010.c:48]: Yeah it looks like mis-named. Let's use `PAL_OVL` and then rename it as a separate PR

**Review by Xeeynamo — APPROVED** [review]
Just one minor comment on `self->palette = 0x8194;` but everything else looks great!

**Pull request merged**